File: java/src/test/java/com/dddeurope/niuz/fixtures/InMemoryTeasers.java
package com.dddeurope.niuz.fixtures;

import com.dddeurope.niuz.entities.Teaser;
import com.dddeurope.niuz.repositories.TeaserRepository;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class InMemoryTeasers implements TeaserRepository {
    final Map<String, List<Teaser>> teasers = new HashMap<>();

    @Override
    public void save(String page, Teaser teaser) {
        teasers.putIfAbsent(page, new ArrayList<>());
        teasers.get(page).add(teaser);
    }

    @Override
    public List<Teaser> getByPage(String page) {
        return teasers.getOrDefault(page, new ArrayList<>());
    }}

File: dotnet/niuz.application.test/fixtures/InMemoryTeasers.cs
using System.Collections.Generic;
using niuz.application.entities;
using niuz.application.repositories;
using niuz.application.website;

namespace niuz.application.fixtures
{
    public class InMemoryTeasers : ITeaserRepository
    {
        private readonly Dictionary<string, List<Teaser>> pages = new Dictionary<string, List<Teaser>>();

        public void Save(string page, Teaser teaser)
        {
            FindPage(page).Add(teaser);
        }

        public IEnumerable<Teaser> GetByPage(string page)
        {
            return FindPage(page);
        }

        private List<Teaser> FindPage(string page)
        {
            if (!pages.ContainsKey(page)) pages[page] = new List<Teaser>();
            return pages[page];
        }
    }
}